src/pages/Search/AdvancedSearchFilters.tsx

```diff
@@ -1,54 +1,100 @@
     const {translate} = useLocalize();
+    const styles = useThemeStyles();
     const {singleExecution} = useSingleExecution();
     const waitForNavigate = useWaitForNavigation();
 
+    const [searchAdvancedFilters = {}] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
+
     const advancedFilters = useMemo(
         () => [
             {
-                title: getFilterDisplayTitle({}, 'title'),
+                title: getFilterDisplayTitle(searchAdvancedFilters, 'type', translate),
                 description: 'common.type' as const,
                 route: ROUTES.SEARCH_ADVANCED_FILTERS_TYPE,
             },
             {
-                title: getFilterDisplayTitle({}, 'date'),
+                title: getFilterDisplayTitle(searchAdvancedFilters, 'date', translate),
                 description: 'common.date' as const,
                 route: ROUTES.SEARCH_ADVANCED_FILTERS_DATE,
             },
+            {
+                title: getFilterDisplayTitle(searchAdvancedFilters, 'status', translate),
+                description: 'search.filters.status' as const,
+                route: ROUTES.SEARCH_ADVANCED_FILTERS_STATUS,
+            },
         ],
-        [],
+        [searchAdvancedFilters, translate],
     );
 
     return (
-        <View>
-            {advancedFilters.map((item) => {
-                const onPress = singleExecution(waitForNavigate(() => Navigation.navigate(item.route)));
-
-                return (
-                    <MenuItemWithTopDescription
-                        key={item.description}
-                        title={item.title}
-                        description={translate(item.description)}
-                        shouldShowRightIcon
-                        onPress={onPress}
-                    />
-                );
-            })}
+        <View style={[styles.flex1, styles.justifyContentBetween]}>
+            <View>
+                {advancedFilters.map((item) => {
+                    const onPress = singleExecution(waitForNavigate(() => Navigation.navigate(item.route)));
+
+                    return (
+                        <MenuItemWithTopDescription
+                            key={item.description}
+                            title={item.title}
+                            description={translate(item.description)}
+                            shouldShowRightIcon
+                            onPress={onPress}
+                        />
+                    );
+                })}
+            </View>
+            <FormAlertWithSubmitButton
+                buttonText={translate('search.viewResults')}
+                containerStyles={[styles.m4]}
+                onSubmit={() => {
+                    // here set the selected filters as new query and redirect to SearchResults page
+                    // waiting for: https://github.com/Expensify/App/issues/45028 and https://github.com/Expensify/App/issues/45027
+                    Navigation.goBack();
```

Comment:
I think we should also clear the form values when navigating back. We should always populate it from the URL params when opening the form and always clear when closing the filters page.